python_ta/typecheck/errors.py

```diff
@@ -101,6 +102,22 @@ def unaryop_error_message(node: astroid.UnaryOp) -> str:
     )
 
 
+###############################################################################
+# Subscript message
+###############################################################################
+def subscript_error_message(node: astroid.Subscript) -> str:
+    # Accessing an element of a List with an incompatible index type (non-integers)
+    if _gorg(node.value.inf_type.getValue()) == List:
+        return f'You can only access elements of a list using an int. ' \
+               f'You used a {node.slice.inf_type.getValue().__name__}, {node.slice.value.as_string()}.'
+    elif _gorg(node.value.inf_type.getValue()) == Tuple:
+        return f'You can only access elements of a tuple using an int. ' \
+               f'You used a {node.slice.inf_type.getValue().__name__}, {node.slice.value.as_string()}.'
+    elif _gorg(node.value.inf_type.getValue()) == Dict:
+        return f'You tried to access an element of this dictionary using an int, 1; however, the type ' \
```

Comment:
Use a period instead of a semicolon. About the wording of the second part, I'd use the plural form: "However, the type of this dictionary's keys is blah."